db/queries.py

```diff
@@ -1,46 +1,66 @@
 from db.models import Message, User, Chat
+
 from django.db.models import Q, Count, F
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=word))
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(title__startswith="Untitled"))
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(User.objects.filter(
+        message__sent__year=2015
+    ).values_list("first_name", "last_name"))
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(message__sent__year__gt=2020))
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=F("user__first_name")))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return list(User.objects.filter(
+        Q(message__text__startswith="a")
+        | Q(message__text__startswith="m")
+    ))
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return list(Message.objects.filter(
+        Q(is_delivered=True)
+        | Q(user__username__istartswith="admin")
+    ))
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(
+        user__first_name=first_name
+    ).aggregate(Count("id"))["id__count"]
 
 
 def get_top_users_by_number_of_the_messages() -> list[User]:
-    pass
+    return list(User.objects.annotate(
+        num_messages=Count("message__user_id")).order_by("-num_messages"))[:3]
 
 
 def get_last_5_messages_dicts() -> list[dict]:
-    pass
+    result = []
+    for message in Message.objects.order_by(
```

Comment:
I think it would look cleaner if you used list comprehension here and in the next function.

Reply:
Thanks. Changed.
As for second function, in my opinion it would be better for reading if usual "for" is used.